rclcpp/minimal_subscriber/static_wait_set.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021, Apex.AI Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+/* This example creates a subclass of Node and uses static a wait-set based loop to wait and handle
+ * messages from the subscriber */
+
+class MinimalSubscriber : public rclcpp::Node
+{
+public:
+  MinimalSubscriber()
+  : Node("minimal_subscriber")
+  {
+    rclcpp::CallbackGroup::SharedPtr cb_group_waitset = this->create_callback_group(
+      rclcpp::CallbackGroupType::MutuallyExclusive, false);
+    auto options = rclcpp::SubscriptionOptions();
+    options.callback_group = cb_group_waitset;
+    auto subscription_callback = [this](std_msgs::msg::String::UniquePtr msg) {
+        RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg->data.c_str());
+      };
+
+    subscription_ = this->create_subscription<std_msgs::msg::String>(
+      "topic",
+      10,
+      subscription_callback,
+      options);
+
+    wait_set_ptr_ = std::make_shared<rclcpp::StaticWaitSet<1, 0, 0, 0, 0, 0>>(
```

Comment:
Yeah, I tried that at first. The main problem was passing the subscription options with the new callback-group. Here it would be useful to have a subscription creation method which creates directly a subscription passing a callback-group and and without needing a callback. I'm not sure if this is feasible or not.

Reply:
Solved by using a lambda